### PR TITLE
Updating WSL bug template to use GitHub Issue Form

### PR DESCRIPTION
**Pull request by adrastogi**
There's a new feature that GitHub is working on called [Issue Forms](https://gh-community.github.io/issue-template-feedback/structured/) which allows for stronger templating around bug reports. The Windows Terminal team has had this feature enabled in their repo for a few weeks now, and have expressed positive feedback about the change, so we're going to try it out on the WSL repo.

The template is written in YAML, and you can preview how it will look [here](https://github.com/microsoft/WSL/blob/issue-form/.github/ISSUE_TEMPLATE/Bug_Report.yaml). 

Note: the `issue_body` key was very recently deprecated, and is flagged as such in the preview of the form. The key will still be around for a few more weeks, so I will issue a subsequent PR to move off of it once I understand the replacement options.

### CODE REVIEW

**Review by benhillis — APPROVED** [review]
Looks good to me!

File: .github/ISSUE_TEMPLATE/Bug_Report.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      Please run `cmd.exe /c ver` to get the build of Windows you are on.
+    placeholder: "Microsoft Windows [Version 10.0.19042.867]"
+  validations:
+    required: true
+
+- type: input
+  attributes:
+    label: WSL Version
+    description: |
+      Tell us whether the issue is on WSL 2 and/or WSL 1; you can do this by running `cat /proc/version` to check the kernel version.
```
Comment (craigloewen-msft) [inline .github/ISSUE_TEMPLATE/Bug_Report.yaml:28]: I'd recommend that we change this to: "You can tell your WSL version by running `wsl.exe -l -v`.

**Comment by craigloewen-msft** [comment]
I recommended one small change on instructing users how to check their WSL version, but other than that looks great! Merge away :)

**Review by craigloewen-msft — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/Bug_Report.yaml (hunk 1 above)
Comment (adrastogi) [reply .github/ISSUE_TEMPLATE/Bug_Report.yaml]: Great suggestion, fixed!

**Pull request merged**